pkg/apis/componentconfig/types.go

```diff
@@ -189,71 +193,65 @@ type KubeControllerManagerConfiguration struct {
 	// ServiceControllerConfiguration holds configuration for ServiceController
 	// related features.
 	ServiceController ServiceControllerConfiguration
-
-	// Controllers is the list of controllers to enable or disable
-	// '*' means "all enabled by default controllers"
-	// 'foo' means "enable 'foo'"
-	// '-foo' means "disable 'foo'"
-	// first item for a particular name wins
-	Controllers []string
-	// externalCloudVolumePlugin specifies the plugin to use when cloudProvider is "external".
-	// It is currently used by the in repo cloud providers to handle node and volume control in the KCM.
-	ExternalCloudVolumePlugin string
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 type CloudControllerManagerConfiguration struct {
 	metav1.TypeMeta
 
-	// CloudProviderConfiguration holds configuration for CloudProvider related features.
-	CloudProvider CloudProviderConfiguration
-	// DebuggingConfiguration holds configuration for Debugging related features.
-	Debugging apiserverconfig.DebuggingConfiguration
-	// GenericComponentConfiguration holds configuration for GenericComponent
+	// GenericControllerManagerConfiguration holds configuration for GenericComponent
 	// related features both in cloud controller manager and kube-controller manager.
-	GenericComponent GenericComponentConfiguration
+	Generic GenericControllerManagerConfiguration
 	// KubeCloudSharedConfiguration holds configuration for shared related features
 	// both in cloud controller manager and kube-controller manager.
 	KubeCloudShared KubeCloudSharedConfiguration
+
 	// ServiceControllerConfiguration holds configuration for ServiceController
 	// related features.
 	ServiceController ServiceControllerConfiguration
 	// NodeStatusUpdateFrequency is the frequency at which the controller updates nodes' status
 	NodeStatusUpdateFrequency metav1.Duration
 }
 
-type CloudProviderConfiguration struct {
-	// Name is the provider for cloud services.
-	Name string
-	// cloudConfigFile is the path to the cloud provider configuration file.
-	CloudConfigFile string
-}
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type GenericControllerManagerConfiguration struct {
```

Comment:
Can you add doc here describing what you envision this used for.  Some aspects look generic to any controller like port, address, client connection, resync.  Others look specific `UseServiceAccountCredentials` is a pretty special thing for the kube-controller-manager since it manages many things in one process and can't use a service account to do it directly.